### PR TITLE
Dobe search into

### PR DESCRIPTION
**Pull request by dobe**

### CODE REVIEW

File: src/main/java/crate/elasticsearch/action/searchinto/parser/SearchIntoParser.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                throw new SearchParseException(context,
+                        "SearchInto field [" + field + "] does not exist in " +
+                                "the mapping");
+            }
+        }
+
+    }
+
+    /**
+     * Main method of this class to parse given payload of _export action
```
Comment (quodt) [inline src/main/java/crate/elasticsearch/action/searchinto/parser/SearchIntoParser.java]: _searchinto

File: src/main/java/crate/elasticsearch/module/searchinto/SearchIntoModule.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+public class SearchIntoModule extends AbstractModule {
+
+
+    @Override
+    protected void configure() {
+        bind(TransportSearchIntoAction.class).asEagerSingleton();
+
+        bind(SearchIntoParser.class).asEagerSingleton();
+        //bind(SearchIntoer.class).asEagerSingleton();
```
Comment (quodt) [inline src/main/java/crate/elasticsearch/module/searchinto/SearchIntoModule.java]: remove

File: src/main/java/crate/elasticsearch/action/searchinto/TransportSearchIntoAction.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+    @Override
+    protected ClusterBlockException checkGlobalBlock(ClusterState state,
+            SearchIntoRequest request) {
+        return state.blocks().globalBlockedException(ClusterBlockLevel.READ);
+    }
+
+    @Override
+    protected ClusterBlockException checkRequestBlock(ClusterState state,
+            SearchIntoRequest exportRequest, String[] concreteIndices) {
```
Comment (quodt) [inline src/main/java/crate/elasticsearch/action/searchinto/TransportSearchIntoAction.java]: maybe rename variable to something different than "exportRequest"

File: src/main/java/crate/elasticsearch/action/searchinto/parser/SearchIntoParser.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import org.elasticsearch.search.SearchParseException;
+import org.elasticsearch.search.fetch.FetchPhase;
+import org.elasticsearch.search.fetch.explain.ExplainParseElement;
+import org.elasticsearch.search.query.QueryPhase;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Parser for payload given to _export action.
```
Comment (quodt) [inline src/main/java/crate/elasticsearch/action/searchinto/parser/SearchIntoParser.java]: _searchinto

File: src/main/java/crate/elasticsearch/rest/action/admin/searchinto/RestSearchIntoAction.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            RestController controller) {
+        super(settings, client);
+        controller.registerHandler(POST, "/_search_into", this);
+        controller.registerHandler(POST, "/{index}/_search_into", this);
+        controller.registerHandler(POST, "/{index}/{type}/_search_into", this);
+    }
+
+    public void handleRequest(final RestRequest request,
+            final RestChannel channel) {
+        SearchIntoRequest exportRequest = new SearchIntoRequest(
```
Comment (quodt) [inline src/main/java/crate/elasticsearch/rest/action/admin/searchinto/RestSearchIntoAction.java]: variable naming

File: src/main/java/crate/elasticsearch/module/searchinto/SearchIntoModule.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        MapBinder<String, WriterCollectorFactory> collectorBinder
+                = MapBinder.newMapBinder(binder(),
+                String.class, WriterCollectorFactory.class);
+
+        collectorBinder.addBinding(BulkWriterCollector.NAME).toProvider(FactoryProvider
+                .newFactory(WriterCollectorFactory.class,
+                        BulkWriterCollector.class));
+
+
+//                .to(BulkWriterCollector.class);
```
Comment (quodt) [inline src/main/java/crate/elasticsearch/module/searchinto/SearchIntoModule.java]: remove

File: src/main/java/crate/elasticsearch/rest/action/admin/searchinto/RestSearchIntoAction.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                    BytesReference querySource = RestActions.parseQuerySource(
+                            request);
+                    if (querySource != null) {
+                        exportRequest.source(querySource, false);
+                    }
+                }
+            }
+            exportRequest.routing(request.param("routing"));
+            exportRequest.types(splitTypes(request.param("type")));
+            exportRequest.preference(request.param("preference", "_primary"));
```
Comment (quodt) [inline src/main/java/crate/elasticsearch/rest/action/admin/searchinto/RestSearchIntoAction.java]: do we really force _primary as default for this endpoint?

**Comment by quodt** [comment]
sonst ack

File: src/main/java/crate/elasticsearch/rest/action/admin/searchinto/RestSearchIntoAction.java (hunk 7 above)
Comment (dobe) [inline src/main/java/crate/elasticsearch/rest/action/admin/searchinto/RestSearchIntoAction.java]: jep, because otherwise it would write twice

**Pull request merged**